index.d.ts

```diff
@@ -6,14 +6,66 @@ export = React;
 export as namespace React;
 
 declare namespace React {
+  type Props = {[key: string]: any};
+  type State = Props;
+  type Context = Props;
+
   export class Component {
-    state: any;
-    props: any;
-    setState: Function;
-    constructor(props: any);
-    static propTypes: any;
-    static defaultProps: any;
+    context: Context;
+    props: Props;
+    state: State;
+    setState: (partialState: State, cb?: Function) => void;
+    constructor(props: Props, context?: Context);
   }
+
+  export class PureComponent extends Component {}
+
   function createElement(...args:any[]): any;
-  var PropTypes : any;
+
+  var Children: ReactChildren;
+  var PropTypes: ReactPropTypes;
+
+  //
+  // React.PropTypes
+  // ----------------------------------------------------------------------
```

Comment:
nit: Might be a good idea to add where these defs were copied from to this comment

Reply:
ah good call - done! https://github.com/Clever/minimal-react-typings/pull/3/commits/dd6c0f219d4b1843d5733112c301024b0e6e0f9b